spec/components/product_component_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe ProductComponent, type: :component do
+  let(:product) { create(:simple_product) }
+
+  describe 'unit' do
+    before do
+      render_inline(
+        ProductComponent.new(
+          product: product, columns: [{ label: "Unit", value: "unit", sortable: false }]
+        )
+      )
+    end
+
+    it 'concatenates the unit value and the unit description' do
+      expect(page.find('.unit')).to have_content '1.0 weight'
+    end
+  end
+
+  describe 'category' do
+    let(:product) do
+      product = create(:simple_product)
+      product.taxons << create(:taxon, name: 'random')
+
+      product
+    end
+
+    before do
+      render_inline(
+        ProductComponent.new(
+          product: product, columns: [{ label: "Category", value: "category", sortable: false }]
+        )
+      )
+    end
+
+    it "joins the categories' name" do
+      expect(page.find('.category')).to have_content(product.taxons.map(&:name).join(', '))
```

Comment:
It's better to hardcode the expected result as a string here. In that case you'd need to pre-define all of the product's taxons. 
`±` Can you create a second taxon above, and pass both in to the `create(:simple_product)` call? I think this would prevent the default taxon being associated.

Reply:
Done. I had to use `product.taxons = taxons` as just passing it to `create(:simple_product)` would still add another default taxon. I think it's due to how the factory is set up but I didn't really dig into it.